internal/services/pubkey_service.go

```diff
@@ -99,6 +117,48 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkeyResource, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
```

Comment:
Ok errors are confusing, I have to say. Not sure how to improve this, @ezr-ondrej? Anyway, when you need to acquire DAO reference, use the InitializeDaoError:

```suggestion
		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err))
```